### DISCUSSION TITLE
GetStreamUri not working for all devices?

### DISCUSSION THREAD

**Discussion by bhavin50** [onvif/testspecs#29] [Q&A] [answered]
Hello,
I am going to test Onvif Client using CTT. In my development I found an issue to get URI.
I want to get URI to play media stream from four ONVIF supported cameras(DH-IPC-HFW2831T-ZAS,QNO-7080R,AXIS P1365 Mk II,DS-2CD2821G0). Basic code(python) is as below.

def get_URI(ips,passw):
    
    mycam = ONVIFCamera(ips, 80, 'admin', passw,'path to /wsdl/')

    media_service = mycam.create_media_service()
    profiles = media_service.GetProfiles()
    token = profiles[0].token
    
    try:
        mediaUri =  media_service.GetStreamUri({'StreamSetup': {'Stream': 'RTP_unicast', 'Transport' : {'Protocol': 'UDP'}} , 'ProfileToken': token})      
    except Exception as e:
        print("Exception in GetStreamUri ",' -- ',e)

Among these four cameras, I can get URI from one camera (Dauha - DH-IPC-HFW2831T-ZAS)only. Anyone can help me why others are not showing? Exceptions for the remaining three cameras are as under.

[1] Unknown error: Validation constraint violation: data type mismatch  in element 'ns1:Stream'
[2] Unknown error: Argument value invalid

Are there any device settings required to change?

**Comment by davidmneedham** [comment, marked as answer]
It looks like this issue may be related to a [third-party Python implementation](https://github.com/quatanium/python-onvif). This is not related to ONVIF specifications.
